OpenAI_API/APIAuthentication.cs

```diff
@@ -9,7 +9,7 @@ namespace OpenAI_API
 	/// <summary>
 	/// Represents authentication to the OpenAPI API endpoint
 	/// </summary>
-	public class APIAuthentication
+	public class APIAuthentication : IAPIAuthentication
```

Comment:
If `APIAuthentication` is not meant to be inherited please add the keyword `sealed` so it's explicitly defined. This comes with some performance benefits: see this PR from the dotnet runtime: https://github.com/dotnet/runtime/issues/49944